extensions/XeroName/Deltatime.js

```diff
@@ -0,0 +1,223 @@
+
+  if (!Scratch.extensions.unsandboxed) {
+    throw new Error('Deltatime by XeroName : This extension must run unsandboxed');
+  }
+
+  const vm = Scratch.vm;
+  const cast = Scratch.Cast;
+
+  let maxFiltStren = 907; // Max value of Filter Strength
+  let filtStren = 8; // Default value of Filter Strength = 8
+  let calcStandFPS = 30; // Standard value of Calculator
+
+  let fT = 0, last = performance.now(), rawFPS, vmFPS, vmDt;
+  const oldStep = vm.runtime._step;
+
+  vm.runtime._step = function () {
+    oldStep.call(this);
+    const now = performance.now();
+    let thisFT = now - last;
+    fT += (thisFT - fT) / filtStren;
+
+    rawFPS = 1000 / fT;
+
+    if (rawFPS < 0.5) {
+      vmFPS = rawFPS.toFixed(2);
+    } else {
+      vmFPS = Math.round(rawFPS);
+    }
+
+    if (vmFPS > 0) {
+      vmDt = 1 / vmFPS; // At normal situation
+    } else {
+      vmDt = 0; // Prevent situation of "DIV/0"
+    }
+    last = now;
+  };
+
+  class Dt {
+    getInfo() {
+      return {
+        id: 'dtbyxeroname',
+        name: 'Deltatime',
+        docsURI: 'https://xeroname.github.io/dtbyxeroname/',
+        color1: '#333333',
+        color2: '#444444',
+        color3: '#ffffff',
+        menuIconURI: icon,
+        blocks: [
+          {
+            opcode: 'getDt_vm',
+            blockType: Scratch.BlockType.REPORTER,
+            text: 'ΔT'
+          },
+          {
+            opcode: 'getFPS_vm',
+            blockType: Scratch.BlockType.REPORTER,
+            text: 'FPS'
+          },
+          {
+            opcode: 'getTimeRes',
+            blockType: Scratch.BlockType.REPORTER,
+            text: 'Get [TYPE] of [TARGET]',
+            arguments: {
+              TYPE: {
+                type: Scratch.ArgumentType.STRING,
+                menu: 'type_menu'
+              },
+
+              TARGET: {
+                type: Scratch.ArgumentType.STRING,
+                menu: 'target_menu'
+              }
+            },
+            hideFromPalette: true
```

Comment:
if no projects use this then we can remove it, same for next block

Reply:
I will remove all of the hidefrompalette blocks.